### ISSUE TITLE
LPC decoding details missing

### ISSUE THREAD

**Issue by fweimer** [ietf-wg-cellar/flac-specification#88]
I think some of the LPC decoding details should be included in the specification:

* Apparently, it is expected that the decoding happens as if with infinite precision.
* Is there an expectation that it is possible to implement subset streams with just an 32x32→32 multiplier if the decoded stream is 16 bit?
* What does it mean if the *quantized linear predictor coefficient shift* in `SUBFRAME_LPC` is negative? I don't think the reference implementation performs a left shift in this case.
* I don't think it is obvious where to apply the quantiziation shift.

It may make sense to include an explicit formula for LPC decoding, so that the it's clear whether there is a minus sign involved or not.

**Comment by ktmf01** [comment]
@fweimer, I was reviewing whether all points you mention are included in the current PRs, and I think they are all accounted for. The negative shift is explicitly forbidden in #97, where to apply the shift is also in #97, the 32-bit math is explained in #124 and the explicit formula is in the (informative appendix) #120

The only thing I'm not sure of is the infinite precision. What do you mean by that?

**Comment by fweimer** [comment]
With *infinite precision* I meant that the decoder has to make sure that the linear predictor computations do not overflow by choosing a sufficiently large data type. It was not clear to me whether there was a rule that encoders had to make sure to stay within a specific numerical range (depending on sample width etc.). #124 makes it clear that decoders have to assume the worst case (what I called infinite precision).

So I think everything I raised here is being addressed. Thanks!
